pkg/agents/team4/finalAgent/trustScoreUpdates.go

```diff
@@ -0,0 +1,117 @@
+	if a.PlatformOnFloor() && len(a.params.responseMessages) > 0 { // Check if there are any responses messages.
+		for i := 0; i < len(a.params.responseMessages); i++ { // Iterate through each response message
+			respMsg := a.params.responseMessages[i]
+			resMsg, ok := respMsg.(messages.ResponseMessage)
+			if !ok {
+				err := fmt.Errorf("ResponseMessage type assertion failed")
+				fmt.Println(err.Error())
+			} else {
+				for j := 0; j < len(a.params.sentMessages); j++ { // Iterate through each sent message
+					sentMsg := a.params.sentMessages[j]
+
+					if resMsg.RequestID() == sentMsg.ID() { // Find the corresponding response message that's been sent
+						a.params.sentMessages = remove(a.params.sentMessages, j) // Remove the accessed response/sent messages from memory
+						a.params.responseMessages = remove(a.params.responseMessages, i)
+						a.Log("Team4 received a message", infra.Fields{"sender_uuid": msg.ID(), "sentmessage_uuid": sentMsg.ID()})
+
+						if sentMsg.MessageType() == messages.RequestLeaveFood && a.Floor()-msg.SenderFloor() == 1 { //TODO: theres now target floors and not directions anymore
+							a.Log("Team4 reponse message received", infra.Fields{"sentMsg_Type": sentMsg.MessageType()})
+							reqMsg, ok := sentMsg.(messages.RequestMessage)
+							if !ok {
+								err := fmt.Errorf("RequestMessage type assertion failed")
+								fmt.Println(err.Error())
+							} else if food.FoodType(reqMsg.Request()) <= a.CurrPlatFood() {
+								a.AddToGlobalTrust(a.params.coefficients[1])
+							} else if food.FoodType(reqMsg.Request()) > a.CurrPlatFood() {
+								a.SubFromGlobalTrust(a.params.coefficients[1])
+								a.Log("Team4: For Requested Food to Leave greater than Food on platform", infra.Fields{"Request_amt": reqMsg.Request(), "Food_on_our_level": a.CurrPlatFood(), "global_trust": a.params.globalTrust})
+							}
+						}
+						break
+					}
+				}
+
+			}
+
+		}
+	} else if a.params.lastFoodTaken+a.CurrPlatFood() != a.params.lastPlatFood {
+		for i := 0; i < len(a.params.responseMessages); i++ { // Iterate through each response message
+			respMsg := a.params.responseMessages[i]
+			resMsg, ok := respMsg.(messages.ResponseMessage)
+			if !ok {
+				err := fmt.Errorf("ResponseMessage type assertion failed")
+				fmt.Println(err.Error())
+			} else {
+				for j := 0; j < len(a.params.sentMessages); j++ { // Iterate through each sent message
+					sentMsg := a.params.sentMessages[j]
+
+					if resMsg.RequestID() == sentMsg.ID() { // Find the corresponding response message that's been sent
+						a.params.sentMessages = remove(a.params.sentMessages, j) // Remove the accessed response/sent messages from memory
+						a.params.responseMessages = remove(a.params.responseMessages, i)
+						a.Log("Team4 received a message", infra.Fields{"sender_uuid": msg.ID(), "sentmessage_uuid": sentMsg.ID()})
+
+						if sentMsg.MessageType() == messages.RequestTakeFood && a.Floor()-msg.SenderFloor() == -1 {
```

Comment:
If this is the only line that has a change in these 35 lines of code (line 49-81 and 13-46) I think you should do a function that passes the message type instead of these two chunks of code. It seems over complicated.